tests/difference/core/quint_model/libraries/extraSpells.qnt

```diff
@@ -0,0 +1,259 @@
+  /// - @returns the value associated with the key, if __key is
+  ///   present in the map, and __default otherwise
+  pure def getOrElse(__map: a -> b, __key: a, __default: b): b = {
+    if (__map.has(__key)) {
+      __map.get(__key)
+    } else {
+      __default
+    }
+  }
+
+  run getOrElseTest = all {
+    assert(Map(2 -> 3, 4 -> 5).getOrElse(2, 0) == 3),
+    assert(Map(2 -> 3, 4 -> 5).getOrElse(7, 11) == 11),
+  }
+
+  /// Remove a map entry.
+  ///
+  /// - @param __map a map to remove an entry from
+  /// - @param __key the key of an entry to remove
+  /// - @returns a new map that contains all entries of __map
+  ///          that do not have the key __key
+  pure def mapRemove(__map: a -> b, __key: a): a -> b = {
+    __map.keys().setRemove(__key).mapBy(__k => __map.get(__k))
+  }
+
+  run mapRemoveTest = all {
+    assert(Map(3 -> 4, 7 -> 8) == Map(3 -> 4, 5 -> 6, 7 -> 8).mapRemove(5)),
+    assert(Map() == Map().mapRemove(3)),
+  }
+
+    /// Removes a set of map entry.
+    ///
+    /// - @param __map a map to remove an entry from
+    /// - @param __keys a set of keys to remove from the map
+    /// - @returns a new map that contains all entries of __map
+    ///          that do not have a key in __keys
+    pure def mapRemoveAll(__map: a -> b, __keys: Set[a]): a -> b = {
+        __map.keys().exclude(__keys).mapBy(__k => __map.get(__k))
+    }
+
+    run mapRemoveAllTest =
+        val m = Map(3 -> 4, 5 -> 6, 7 -> 8)
+        all {
+            assert(m.mapRemoveAll(Set(5, 7)) == Map(3 -> 4)),
+            assert(m.mapRemoveAll(Set(5, 99999)) == Map(3 -> 4, 7 -> 8)),
+        }
+
+    //// Returns a list of all elements of a set.
+    ////
+    //// - @param __set a set
+    //// - @returns a list of all elements of __set
+    pure def toList(__set: Set[a]): List[a] = {
+        __set.fold(List(), (__l, __e) => __l.append(__e))
+    }
+
+    //// Returns a set of the elements in the list.
+    ////
+    //// - @param __list a list
+    //// - @returns a set of the elements in __list
+    pure def toSet(__list: List[a]): Set[a] = {
+        __list.foldl(Set(), (__s, __e) => __s.union(Set(__e)))
+    }
+
+    run toListAndSetTest =
+    all {
+        assert(Set(3, 2, 1).toList().toSet() == Set(1, 2, 3)),
+        assert(List(2,3,1).toSet() == Set(1, 2, 3)),
+        assert(List(2,3,1).toSet() == List(3,2,1).toSet()),
+        assert(toList(Set()) == List()),
+        assert(toSet(List()) == Set())
+    }
+
+    pure def add(__set: Set[a], elem: a): Set[a] = {
+        __set.union(Set(elem))
+    }
+
+    pure def Values(__map: a -> b): Set[b] = {
+        __map.keys().fold(Set(), (__s, __k) => __s.add(__map.get(__k)))
+    }
+    run ValuesTest =
+    all {
+        assert(Values(Map(1 -> 2, 3 -> 4)) == Set(2, 4)),
+        assert(Values(Map()) == Set())
+    }
+
+    // Returns the maximal element of the set.
+    // If the set is empty, the function call will fail at runtime.
+    pure def Max(__set: Set[int]): int = {
```

Comment:
1. Why` are `Max`, `Min` capitalized in comparison to other `def`s?
2. (nit) You could use `MaxBy` to define `Max` with something like:
``` pure def Max(__set: Set[int]): int = MaxBy(__set, a => a)```

Reply:
Unified this, thanks